### PR TITLE
groups: better roles handling, ensure valid states

### PR DESCRIPTION
**Pull request by arthyn**
This ensures that we recheck permissions upon any relevant changes. It also tries to clean up any role state that is currently invalid. Also makes sure the frontend doesn't actually send "members" as a role and cleans up a few things there.

Fixes LAND-616
Fixes LAND-621

### CODE REVIEW

File: desk/app/diary.hoon [hunk 1]
```diff
@@ -465,33 +465,38 @@
...
-    =/  di  (di-abed:di-core:co flag)
-    di-abet:di-recheck:di
+      [%cabal * %del *]
+    =/  =sect:g  (slav %tas p.diff)
+    %+  recheck-perms  affected
+    `(~(gas in *(set sect:g)) ~[p.diff])
   ==
 ::
+++  recheck-perms
+  |=  [affected=(list flag:d) sects=(unit (set sect:g))]
```
Comment (midsum-salrux) [inline desk/app/diary.hoon:494]: NB: would calling this with an empty set in the null case be clearer than using a unit or is there a semantic difference between the two?

**Review by midsum-salrux — APPROVED** [review]
LGTM with one non-blocking question

File: ui/src/groups/GroupAdmin/GroupRoles.tsx [hunk 2]
```diff
@@ -370,7 +370,8 @@ export default function GroupRoles({ title }: { title: string }) {
               >
                 remove all members
               </Link>{' '}
-              from this role before it can be deleted.
+              from this role and remove it from all channels before it can be
+              deleted.
```
Comment (jamesacklin) [inline ui/src/groups/GroupAdmin/GroupRoles.tsx:374]: Ah-ha! Thanks.

File: desk/app/diary.hoon (hunk 1 above)
Comment (arthyn) [reply desk/app/diary.hoon:494]: hmm yeah we probably could use an empty set here

**Pull request merged**